lib/neo4j/active_node/query/query_proxy_enumerable.rb

```diff
@@ -14,6 +14,8 @@ def each(node = true, rel = nil, &block)
         end
 
         def result(node = true, rel = nil)
+          return [].freeze if @start_object && @start_object.new_record?
```

Comment:
Have a look as well how I tried to solve the problem there (like you using the `@start_object). And like you I covered the creation case only, but not the update case.